storio-sqlite-annotations-processor/src/main/java/com/pushtorefresh/storio/sqlite/annotations/processor/generate/GetResolverGenerator.java

```diff
@@ -116,4 +101,88 @@ private MethodSpec createMapFromCursorMethodSpec(@NotNull StorIOSQLiteTypeMeta s
+        final MethodSpec.Builder builder = MethodSpec.methodBuilder("mapFromCursor")
+                .addJavadoc("{@inheritDoc}\n")
+                .addAnnotation(Override.class)
+                .addAnnotation(ANDROID_NON_NULL_ANNOTATION_CLASS_NAME)
+                .addModifiers(PUBLIC)
+                .returns(storIOSQLiteTypeClassName)
+                .addParameter(ParameterSpec.builder(ClassName.get("android.database", "Cursor"), "cursor")
+                        .addAnnotation(ANDROID_NON_NULL_ANNOTATION_CLASS_NAME)
+                        .build())
+                .addCode("\n");
+
+        // We don't know the name of the variable, just the type. So just use generic parameter names
+        int paramCount = 1;
+        final StringBuilder paramsBuilder = new StringBuilder();
+        paramsBuilder.append("(");
+        boolean first = true;
+        for (final StorIOSQLiteColumnMeta columnMeta : storIOSQLiteTypeMeta.columns.values()) {
+            final String columnIndex = "cursor.getColumnIndex(\"" + columnMeta.storIOColumn.name() + "\")";
+
+            final JavaType javaType = columnMeta.javaType;
+
+            final String getFromCursor = getFromCursorString(columnMeta, javaType, columnIndex);
+
+            final TypeName name = TypeName.get(((ExecutableElement) columnMeta.element).getReturnType());
+
+            final boolean isBoxed = javaType.isBoxedType();
+            if (isBoxed) { // otherwise -> if primitive and value from cursor null -> fail early
+                builder.addStatement("$T param" + paramCount + " = null", name);
+                builder.beginControlFlow("if(!cursor.isNull($L))", columnIndex);
+                builder.addStatement("param" + paramCount + " = cursor.$L", getFromCursor);
+                builder.endControlFlow();
+            } else {
+                builder.addStatement("$T param" + paramCount + " = cursor.$L", name, getFromCursor);
+            }
+
+            if (!first) {
+                paramsBuilder.append(" ,");
+            }
+            first = false;
+            paramsBuilder.append("param" + paramCount);
+
+            paramCount++;
+        }
+        paramsBuilder.append(")");
+        builder.addCode("\n");
+
+        if (storIOSQLiteTypeMeta.creator.getKind() == ElementKind.CONSTRUCTOR) {
+            builder.addStatement("$T object = new $T" + paramsBuilder.toString(), storIOSQLiteTypeClassName, storIOSQLiteTypeClassName);
+        } else {
+            builder.addStatement("$T object = $T." + storIOSQLiteTypeMeta.creator.getSimpleName() + paramsBuilder.toString(), storIOSQLiteTypeClassName, storIOSQLiteTypeClassName);
+        }
+
+        return builder
+                .addCode("\n")
+                .addStatement("return object")
+                .build();
+    }
+
+    @NotNull
+    private String getFromCursorString(@NotNull StorIOSQLiteColumnMeta columnMeta, @NotNull JavaType javaType, @NotNull String columnIndex) {
```

Comment:
Probably in `GetResolverGenerator` in content resolver package. As I told earlier I haven't touched it yet.